squidDrone/tests/propulsion/motorbuilder_test.cpp

```diff
@@ -0,0 +1,48 @@
+#include <memory>
+#include <utility>
+
+#include "gtest/gtest.h"
+#include "motor_builder.hpp"
+#include "motor_mock.hpp"
+#include "propulsion_hardware_config_mock.hpp"
+#include "stm32g4xx_hal_tim.h"
+
+namespace {
+
+class MotorBuilderTest : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    timer.test_member = 42;
+    config.motor_type = types::MotorType::LETODAR_2204;
+    config.esc_type = types::EscType::LITTLE_BEE_20_A;
+    config.timer = &timer;
+    config.channel = TIM_CHANNEL_2;
+  }
+
+  propulsion::PropulsionHardwareConfig config;
+  TIM_HandleTypeDef timer;
+};
+
+TEST_F(MotorBuilderTest, creates_correct_object) {
+  auto motor_object = propulsion::MotorBuilder::Create(config);
+  ASSERT_EQ(motor_object->is_created_, true);
+}
+
+TEST_F(MotorBuilderTest, channel_is_not_in_allowed_channels) {
```

Comment:
What happens if ESC is not allowed type?
You check for motor type and channel type.

Reply:
Fixed in [this commit](https://github.com/elheck/Squiddrone/pull/230/commits/e04970a97abe893523fc4550125cd271e221a661).
Added another test